engine/schema/src/main/java/com/cloud/host/dao/HostDaoImpl.java

```diff
@@ -480,6 +482,25 @@ public Integer countAllByTypeInZone(long zoneId, Type type) {
         return listBy(sc);
     }
 
+    @Override
+    public boolean checkHostServiceOfferingTags(HostVO host, ServiceOffering serviceOffering){
+        if (host == null) {
+            return false;
+        }
+        if (serviceOffering == null) {
+            return false;
+        }
+        if (Strings.isNullOrEmpty(serviceOffering.getHostTag())) {
+            return true;
+        }
+
+        List<String> serviceOfferingTags = Arrays.asList(serviceOffering.getHostTag().split(","));
+        if(host.getHostTags() != null && host.getHostTags().containsAll(serviceOfferingTags)){
+            return true;
+        }
+        return false;
+    }
```

Comment:
trying to isolate this check is fine but the DAO is for DB interaction, and this is really business logic. personally I would make it a default method for the Host interface, but a member of HostVO could also be.
Can you (shortly) explain why you put it here, @DK101010 ?

Reply:
Hi,
In the HostDaoImpl there are a couple of other methods that are check host tags, therefore i thought will be good to add this also here. So will be encapsulate common logic in one class, but my experience is still not very well in CS and when it destroy other guidelines then will be also ok to move this method. For me sounds HostVO good, because it  seems to be only relevant for this object.

Reply:
as said I prefer a default method on the interface but it maight require undesirable imports so HostVO is good for my part.

Reply:
will you be moving this method @DK101010 ?

Reply:
is moved ;)